File: ATB_1.1_src/src/main/java/edu/stanford/isis/atb/domain/template/Interval.java
/*
*  Copyright Northwestern University
*  Copyright Stanford University (ATB 1.0 and ATS 1.0)
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/



package edu.stanford.isis.atb.domain.template;

import javax.validation.constraints.NotNull;

import org.hibernate.validator.constraints.NotEmpty;
import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Root;

/**
 * @author Vitaliy Semeshko
 */

@Root(name="Interval", strict=false)
public class Interval extends AbstractRemovableElement {

	@Attribute(required=false)
	private double minValue;
	
	@Attribute(required=false)
	private double maxValue;
	
	@NotNull(message = "{interval.minOperator.notNull}")
	@Attribute(required=false)
	private Operator minOperator;
	
	@NotNull(message = "{interval.maxOperator.notNull}")
	@Attribute(required=false)
	private Operator maxOperator;
	
	@NotEmpty(message = "{interval.ucumString.notEmpty}")
	@Attribute(required=false)
	private String ucumString;

	public double getMinValue() {
		return minValue;
	}

	public void setMinValue(double minValue) {
		this.minValue = minValue;
	}

	public double getMaxValue() {
		return maxValue;
	}

	public void setMaxValue(double maxValue) {
		this.maxValue = maxValue;
	}

	public Operator getMinOperator() {
		return minOperator;
	}

	public void setMinOperator(Operator minOperator) {
		this.minOperator = minOperator;
	}

	public Operator getMaxOperator() {
		return maxOperator;
	}

	public void setMaxOperator(Operator maxOperator) {
		this.maxOperator = maxOperator;
	}

	public String getUcumString() {
		return ucumString;
	}

	public void setUcumString(String ucumString) {
		this.ucumString = ucumString;
	}
	
	@Override
	public String toString() {
		return "Interval: min: " + minValue + " max: " + maxValue;
	}
	
}

File: AIMToolkit_v3.0.2_rv11/source/AIMLib/AIMLib/model/Interval.cpp
/*
*  Copyright Northwestern University
*  Copyright Stanford University (ATB 1.0 and ATS 1.0)
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/


#include "../stdafx.h"

#include "../AimEnum.h"

#include "AimBase.h"
#include "CharacteristicQuantification.h"
#include "Interval.h"

#include "../memdebug.h"

using std::string;
using namespace aim_lib;

Interval::Interval(void) : CharacteristicQuantification(C_QUANTIFICATION_INTERVAL_NAME)
{
	_minValue		= 0.0;
	_maxValue		= 0.0;
	_minOperator	= None;
	_maxOperator	= None;
}

Interval::Interval(const Interval& interval) : CharacteristicQuantification(interval)
{
	*this = interval;
}

Interval::~Interval(void)
{
}

Interval& Interval::operator=(const Interval& interval)
{
	if (this != &interval)
	{
		CharacteristicQuantification::operator=(interval);
		_minValue		= interval._minValue;
		_maxValue		= interval._maxValue;
		_minOperator	= interval._minOperator;
		_maxOperator	= interval._maxOperator;
		_ucumString		= interval._ucumString;
	}

	return *this;
}

Interval* Interval::Clone() const
{
	return new Interval(*this);
}

double Interval::GetMinValue() const
{
	return _minValue;
}

double Interval::GetMaxValue() const
{
	return _maxValue;
}

ComparisonOperatorType Interval::GetMinOperator() const
{
	return _minOperator;
}

ComparisonOperatorType Interval::GetMaxOperator() const
{
	return _maxOperator;
}

const string& Interval::GetUcumString() const
{
	return _ucumString;
}

void Interval::SetMinValue(double newVal)
{
	_minValue = newVal;
}

void Interval::SetMaxValue(double newVal)
{
	_maxValue = newVal;
}

void Interval::SetMinOperator(ComparisonOperatorType newVal)
{
	_minOperator = newVal;
}

void Interval::SetMaxOperator(ComparisonOperatorType newVal)
{
	_maxOperator = newVal;
}

void Interval::SetUcumString(const string& newVal)
{
	_ucumString = newVal;
}
